lib/active_model/serializer/adapter.rb

```diff
@@ -2,12 +2,15 @@ module ActiveModel
   class Serializer
     class Adapter
       extend ActiveSupport::Autoload
+      autoload :CacheCheck
       require 'active_model/serializer/adapter/json'
       require 'active_model/serializer/adapter/json_api'
-      autoload :FlattenJson
+      autoload :Attributes
       autoload :Null
       autoload :FragmentCache
 
+      include CacheCheck
```

Comment:
We are using `autoload` and including it in the sequence, shouldn't we go with `require` instead then?